icu4c/source/tools/icuwriteuprops/icuwriteuprops.cpp

```diff
@@ -0,0 +1,207 @@
+// © 2016 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include <iostream>
+#include "toolutil.h"
+#include "uoptions.h"
+#include "cmemory.h"
+#include "charstr.h"
+#include "cstring.h"
+#include "unicode/uchar.h"
+#include "unicode/errorcode.h"
+#include "unicode/uniset.h"
+#include "unicode/putil.h"
+#include "unicode/umutablecptrie.h"
+#include "writesrc.h"
+
+U_NAMESPACE_USE
+
+/*
+ * Global - verbosity
+ */
+UBool VERBOSE = FALSE;
+UBool QUIET = FALSE;
+
+UBool haveCopyright = TRUE;
+UCPTrieType trieType = UCPTRIE_TYPE_SMALL;
+
+void handleError(const ErrorCode& status, const char* context) {
+    if (status.isFailure()) {
+        std::cerr << "Error: " << context << ": " << status.errorName() << std::endl;
+        exit(status.get());
+    }
+}
+
+void dumpBinaryProperty(UProperty uproperty, FILE* f) {
+    ErrorCode status;
+    const char* fullPropName = u_getPropertyName(uproperty, U_LONG_PROPERTY_NAME);
```

Comment:
optional: I would (still) pull the property name getting & printing from both dump functions into the caller.